src/Saunter/AsyncApiMiddleware.cs

```diff
@@ -3,57 +3,57 @@
+        _asyncApiDocumentSerializer = asyncApiDocumentSerializer;
+        _options = options.Value;
+    }
 
-        public AsyncApiMiddleware(RequestDelegate next, IOptions<AsyncApiOptions> options, IAsyncApiDocumentProvider asyncApiDocumentProvider, IAsyncApiDocumentSerializer asyncApiDocumentSerializer)
+    public async Task InvokeAsync(HttpContext context)
+    {
+        if (!context.IsRequestingAsyncApiDocument(_options))
         {
-            _next = next;
-            _asyncApiDocumentProvider = asyncApiDocumentProvider;
-            _asyncApiDocumentSerializer = asyncApiDocumentSerializer;
-            _options = options.Value;
+            await _next(context);
+            return;
         }
 
-        public async Task Invoke(HttpContext context)
+        var prototype = _options.AsyncApi;
+        if (context.TryGetDocument(_options, out var documentName) && !_options.NamedApis.TryGetValue(documentName, out prototype))
         {
-            if (!IsRequestingAsyncApiSchema(context.Request))
-            {
-                await _next(context);
-                return;
-            }
-
-            var prototype = _options.AsyncApi;
-            if (context.TryGetDocument(out var documentName) && !_options.NamedApis.TryGetValue(documentName, out prototype))
-            {
-                await _next(context);
-                return;
-            }
-
-            var asyncApiSchema = _asyncApiDocumentProvider.GetDocument(_options, prototype);
-
-            await RespondWithAsyncApiSchemaJson(context.Response, asyncApiSchema, _asyncApiDocumentSerializer, _options);
+            await _next(context);
+            return;
         }
 
-        private static async Task RespondWithAsyncApiSchemaJson(HttpResponse response, AsyncApiSchema.v2.AsyncApiDocument asyncApiSchema, IAsyncApiDocumentSerializer asyncApiDocumentSerializer, AsyncApiOptions options)
-        {
-            var asyncApiSchemaJson = asyncApiDocumentSerializer.Serialize(asyncApiSchema);
-            response.StatusCode = (int)HttpStatusCode.OK;
-            response.ContentType = asyncApiDocumentSerializer.ContentType;
+        var asyncApiSchema = _asyncApiDocumentProvider.GetDocument(_options, prototype);
 
-            await response.WriteAsync(asyncApiSchemaJson);
-        }
+        await RespondWithAsyncApiSchemaJsonAsync(context.Response, asyncApiSchema, _asyncApiDocumentSerializer);
+    }
 
-        private bool IsRequestingAsyncApiSchema(HttpRequest request)
-        {
-            return HttpMethods.IsGet(request.Method) && request.Path.IsMatchingRoute(_options.Middleware.Route);
-        }
+    private static async Task RespondWithAsyncApiSchemaJsonAsync(
+        HttpResponse response,
+        AsyncApiSchema.v2.AsyncApiDocument asyncApiSchema,
+        IAsyncApiDocumentSerializer asyncApiDocumentSerializer)
+    {
+        var asyncApiSchemaJson = asyncApiDocumentSerializer.Serialize(asyncApiSchema);
```

Comment:
This could be cached instead of serialising every time

Reply:
My apologies but that's what's been used all the time. That's not what this PR is about. It's about enabling others to use plain middlewares in order to be less limited.

That being said, thanks for your review.